pkgs/development/compilers/flutter/flutter.nix

```diff
@@ -30,47 +30,44 @@
 , nspr
 , nss
 , systemd
+, callPackage
 }:
 let
-  drvName = "flutter-${channel}-${version}";
+  repository = callPackage ./repository.nix {
+    inherit src pname version depsSha256;
```

Comment:
here dart has to be passed as well. otherwise overriden dart for a custom flutter version does not work

Reply:
I'm trying to use `dev` with your changes. This requires dart 2.12. So I do:

```nix
        version = "1.26.0-12.0.pre";
        channel = "dev";
        filename = "flutter_linux_${version}-${channel}.tar.xz";
        dartVersion = "2.12.0-133.7";
        dartVersionFull = "2.12.0-133.7.beta";
        mydart = pkgs.dart.override {
          version = dartVersion;
          sources = let
            base = "https://storage.googleapis.com/dart-archive/channels";
            x86_64 = "x64";
          in {
            "${dartVersion}-x86_64-linux" = builtins.fetchurl {
              url =
                "${base}/beta/release/${dartVersionFull}/sdk/dartsdk-linux-${x86_64}-release.zip";
              sha256 =
                "sha256:0lz4p1x99sj39cym702yd0sny5yqvik4hff4x3ji4sawzn8snzpy";
            };

          };
        };

        myflutter = pkgs.flutterPackages.mkFlutter rec {
          inherit version;
          dart = mydart;
          pname = "flutter";
          patches = [ ];
          src = builtins.fetchurl {
            url =
              "https://storage.googleapis.com/flutter_infra/releases/${channel}/linux/${filename}";
            sha256 =
              "sha256:02hldr1hs2h1hhqxh97dc2r2nrymsawrc5l8a78rnp7nfn3sjazs";
          };
          depsSha256 = "0d7vhk6axgqajy2d9ia9lc6awcnz6cc3n04r7hnh7bx4hb0jv0l1";

        };
```

But this fails because of the not passed overriden dart sdk

Reply:
Good catch.

Reply:
Done, can you test again?

Reply:
It builds the dev version but the dev version is broken. I think thats a seperate issue and out of the scope of this one. It works fine the not overriden one.